### PR TITLE
Changed ESP CAN updates to transmit 7 bytes per frame with extended I…

### PR DESCRIPTION
**Pull request by feistjo**
…D, transmit 2 messages per loop iteration

### CODE REVIEW

File: include/can_interface.h [hunk 1]
```diff
@@ -1060,10 +1060,11 @@ class CANRXMessage : public ICANRXMessage
 
     void DecodeSignals(CANMessage message)
     {
-        if (message.id_ != id_)
+        if (message.id_ & id_mask_ != id_ * id_mask_)
```
Comment (kenny-navier) [inline include/can_interface.h]: Should it be:
id_ & id_mask_

File: include/esp_can_update.h [hunk 2]
```diff
@@ -44,37 +45,35 @@ class CANUpdate
...
+    MakeUnsignedCANSignal(uint64_t, 8, 56, 1, 0) update_data_{};
+
     MakeUnsignedCANSignal(uint32_t, 0, 24, 1, 0) update_block_idx_{};
     MakeUnsignedCANSignal(bool, 24, 1, 1, 0) received_len_{};
     MakeUnsignedCANSignal(bool, 25, 1, 1, 0) received_md5_{};
     MakeUnsignedCANSignal(bool, 26, 1, 1, 0) written_{};
 
     CANTXMessage<4> update_progress_message_{
-        can_interface_, kUpdateId + 1, 4, 10, timer_group_, update_block_idx_, received_len_, received_md5_, written_};
+        can_interface_, kUpdateId + 2, 4, 10, timer_group_, update_block_idx_, received_len_, received_md5_, written_};
```
Comment (kenny-navier) [inline include/esp_can_update.h:67]: Why +2?  Can the 2 be given a descriptive constant name?   (more below)

File: include/esp_can_update.h [hunk 3]
```diff
@@ -111,45 +110,50 @@ class CANUpdate
...
+        message_type_,
+        length_signal_group_,
+        md5_signal_group_};
+
+    CANRXMessage<2> update_data_message_{
+        can_interface_,
+        kUpdateId,
+        [this]() {
+            uint64_t data = update_data_;
+            uint32_t index = ((update_data_message_.GetID() & 0x1FFFF800) >> 3) + data_block_index_low_;
```
Comment (kenny-navier) [inline include/esp_can_update.h]: Where does the 0x1FFFF800 come from?  Can you add comment or give value descriptive name.

File: scripts/esp_can_update.py [hunk 4]
```diff
@@ -137,55 +140,77 @@ def on_upload(source, target, env):
...
-                can_bus.send(
-                    can.Message(
-                        arbitration_id=data_message.frame_id, data=data_message_data
+        msgs_in_block = 2
+
+        max_block_written = -1
+        last_update = 0
+        while max_block_written < math.floor((len(firmware_bytes) + 6) / 7) - 1:
+            for j in range(msgs_in_block):
+                data_idx = max_block_written + j + 1  # (i * msgs_in_block) + j
```
Comment (kenny-navier) [inline scripts/esp_can_update.py]: Why is the second half commented out?  If it's dead code, you should remove it.

File: include/esp_can_update.h (hunk 2 above)
Comment (feistjo) [reply include/esp_can_update.h:67]: That's just the ID offset of that message, could be a const but this is the only place it's used and I don't think it's necessary

**Review by kenny-navier — APPROVED** [review]

**Pull request merged**